File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int>topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> hashmap;
        for(int i=0;i<nums.size();i++){
            hashmap[nums[i]]=hashmap[nums[i]]+1;
        }
        multimap <int,int> swapped_map;
        for(const auto& pair:hashmap){
            swapped_map.insert({pair.second,pair.first});
        }
    
        vector<int> answer;
        auto it=swapped_map.rbegin();
        for(int i=0;i<k;i++){
            answer.push_back(it->second);
            ++it;
        }
        return answer;
    } 
};

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        hashmap={}
        for num in nums:
            hashmap[num]=hashmap.get(num,0)+1
        sorted_hashmap = sorted(hashmap.items(), key=lambda x: x[1], reverse=True)
        print(sorted_hashmap)
        answer=[]
        for i in range(k):
            answer.append(sorted_hashmap[i][0])
        return answer
